clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchCollector.java

```diff
@@ -155,7 +155,10 @@ private Fetch<K, V> fetchRecords(final CompletedFetch nextInLineFetch, int maxRe
             log.debug("Not returning fetched records for partition {} since it is no longer assigned", tp);
         } else if (!subscriptions.isFetchable(tp)) {
             // this can happen when a partition is paused before fetched records are returned to the consumer's
-            // poll call or if the offset is being reset
+            // poll call or if the offset is being reset.
+            // It can also happen under the Consumer rebalance protocol, when the consumer changes its subscription.
+            // Until the consumer receives an updated assignment from the coordinator, it can hold assigned partitions
+            // that are not in the subscription anymore, so we make them not fetchable.
```

Comment:
Wasn't one of the cases that was discovered where the coordinator (accidentally) assigned partitions for topics that the consumer wasn't subscribed to?